docs/docs/graphql.md

```diff
@@ -1077,6 +1077,434 @@ export const getWeather = async ({ input }: WeatherInput) {
 }
 ```
 
+## Self-Documenting GraphQL API
+
+RedwoodJS helps you document your GraphQL API by auto-generating commented SDL that will be incorporated into the `graphql.schema` file when generating types and that will be used for GraphiQL and the GraphQL Playground explorer.
+
+If you comment your Prisma schema models, its fields, or enums, the SDL generator will use the comment as the documentation.
+
+If there is no Prisma comment, then the SDL generator will default a comment that you can then edit.
+
+:::note
+If you re-generate the SDL, any custom comments will be overwritten.
+However, if you make those edits in your Prisma schema, the those will be used.
+:::
+
+### Prisma Schema Comments
+
+Your Prisma schema is documented with a comment `///` that precedes:
+
+* the Model name
+* the Enum name
+* each Model field name
+
+```
+/// A blog post.
+model Post {
+  /// The unique identifier of a post.
+  id        Int      @id @default(autoincrement())
+  /// The title of a post.
+  title     String
+  /// The content of a post.
+  body      String
+  /// When the post was created.
+  createdAt DateTime @default(now())
+}
+
+model Contact {
+  /// The unique identifier of a contact.
+  id        Int @id @default(autoincrement())
+  /// The name of the contact.
+  name      String
+  /// The email of the contact.
+  email     String
+  /// The message of the contact.
+  message   String
+  /// When the contact was created.
+  createdAt DateTime @default(now())
+}
+
+/// A list of allowed colors.
+enum Color {
+  RED
+  GREEN
+  BLUE
+}
+```
+
+### SDL Comments
+
+The SDL generator adds comments for:
+
+* Directives
+* Queries
+* Mutations
+* Input Types
+
+Comments in GraphQL schema are enclosed in `"""`.
+
+```
+"""
+Use to check whether or not a user is authenticated and is associated
+with an optional set of roles.
+"""
+directive @requireAuth(roles: [String]) on FIELD_DEFINITION
+
+"""Use to skip authentication checks and allow public access."""
+directive @skipAuth on FIELD_DEFINITION
+
+"""
+Representation of Contact.
+"""
```

Comment:
```suggestion
"""Representation of Contact"""
```

Reply:
And the same for the rest of the comments that need it

Reply:
FYi I tested and it reformats when generating, but good idea. 👍